Gems/Atom/Feature/Common/Code/Source/SkinnedMesh/SkinnedMeshFeatureProcessor.cpp

```diff
@@ -310,6 +333,39 @@ namespace AZ
             return false;
         }
 
+        void SkinnedMeshFeatureProcessor::SetSkinningMatrices(const SkinnedMeshHandle& handle, const AZStd::vector<float>& data)
+        {
+            if (handle.IsValid())
+            {
+                handle->SetSkinningMatrices(data);
+            }
+        }
+
+        void SkinnedMeshFeatureProcessor::SetMorphTargetWeights(
+            const SkinnedMeshHandle& handle, uint32_t lodIndex, const AZStd::vector<float>& weights)
+        {
+            if (handle.IsValid())
+            {
+                handle->SetMorphTargetWeights(lodIndex, weights);
+            }
+        }
+
+        void SkinnedMeshFeatureProcessor::EnableSkinning(const SkinnedMeshHandle& handle, uint32_t lodIndex, uint32_t meshIndex)
+        {
+            if (handle.IsValid())
+            {
+                handle->EnableSkinning(lodIndex, meshIndex);
+            }
+        }
+
+        void SkinnedMeshFeatureProcessor::DisableSkinning(const SkinnedMeshHandle& handle, uint32_t lodIndex, uint32_t meshIndex)
+        {
+            if (handle.IsValid())
+            {
+                handle->DisableSkinning(lodIndex, meshIndex);
+            }
+        }
+
```

Comment:
This is interesting. I know this pattern exists elsewhere, but it makes me ask why we need to call a function on the feature processor to invoke the function on the handle. Seems like an unnecessary step.

Reply:
In this case, the handle is a StableDynamicArray handle, which lets you invoke a function on the handle. But in other feature processors or potentially future versions of this feature, the handle could be something entirely different that is just used to index into an underlying array, which maybe doesn't have that capability. Doing it this way lets that implementation detail of what a 'handle' is vary while remaining consistent in how it is invoked.